uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -259,75 +275,65 @@ INPUTS:
   - $subjectIDsref: subjectIDs hashref
   - $dbhr         : database handle reference
 
-RETURNS: the determined objective, or 0
+RETURNS: the determined objective, or undef
 
 =cut
 
 sub getObjective
 {
-    my ($subjectIDsref, $dbhr) = @_;
+    my ($subjectIDsref, $dbh) = @_;
     my @results = ();
     my $objective = 0;
     my %subjectIDs = %$subjectIDsref;
-    if($subjectIDs{'visitLabel'} =~ /PHA/i or $subjectIDs{'visitLabel'} =~ /TEST/i) {
-	return 0;
-    }
-
-    my $query = "SELECT SubprojectID FROM session WHERE CandID='$subjectIDs{'CandID'}' AND Visit_label='$subjectIDs{'visitLabel'}' AND Active='Y' ORDER BY ID DESC LIMIT 1";
-    my $sth = $${dbhr}->prepare($query) or die "Can't prepare $query: ".$${dbhr}->errstr."\n";
 
-    $sth->execute();
-
-    if($sth->rows > 0) {
-        @results = $sth->fetchrow_array();
-    }
+    my $query = "SELECT SubprojectID "
```

Comment:
should also select the ProjectID of the session since protocol can be restricted per Project or Subproject

Reply:
Since previous comments.

Reply:
same comment. either create a new function to get the ProjectID of the session or add it here.